### PR TITLE
add optimizely x plugin

### LINKED ISSUE #711: support optimizely x
# TL;DR

- [ ] add example and setup AB testing
- [ ] implements plugin module
- [ ] add e2e test for event tracking

### PR DESCRIPTION
**Pull request by shoota**
# TL;DR

Connect to #711 

### How to check

*   checkout this pr.
*   yarn install && yarn build
*   yarn test

### Check list

*   [ ] pass all tests

### CODE REVIEW

File: packages/optimizely_x/src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export class Plugin {
+  private tracker: Agent
+  private isSent: boolean
+  constructor (tracker: Agent) {
+    this.tracker = tracker
+    this.isSent = false
+  }
+
+  public getVariation (global = window as any, interval = 50, max = 3) {
```
Comment (shoota) [inline packages/optimizely_x/src/plugin.ts]: Maybe, 50 mill-sec is too short.

File: packages/optimizely_x/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      campaignIds.forEach((campaignId) => {
+        const experimentId: string = campaignStates[campaignId].experiment.id || ''
+        const variationId: string = campaignStates[campaignId].variation.id || ''
+        if (experimentId !== '' || variationId !== '') {
+          this.tracker.send('event', {
+            eventCategory: name,
+            eventAction: experimentId,
+            eventLabel: variationId
+          })
+          this.isSent = true
```
Comment (shoota) [inline packages/optimizely_x/src/plugin.ts]: Set to true when found one **or more than** variations.

File: packages/optimizely_x/test/optimizely-x.test.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import Agent from '@userdive/agent'
+import * as assert from 'assert'
+import { random } from 'faker'
+import 'mocha'
+import { spy as sinonSpy } from 'sinon'
+import { name, Plugin as Optimizely } from '../src/plugin'
+
+const emulate = (global = window as any, ready = false, match = true) => {
```
Comment (shoota) [inline packages/optimizely_x/test/optimizely-x.test.ts]: inject assumed optimizely x object.
see also: https://developers.optimizely.com/x/solutions/javascript/reference/index.html#function_getdata

File: packages/optimizely_x/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import provide from '@userdive/provider'
+import { Plugin } from './plugin'
+provide('optimizely', Plugin)
```
Comment (shoota) [inline packages/optimizely_x/src/index.ts]: provide name is `optimizely`!  Not include `X`.

File: packages/optimizely_x/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@userdive/optimizely-x-plugin",
```
Comment (shoota) [inline packages/optimizely_x/package.json]: module name include `x`.

File: packages/optimizely_x/src/index.ts (hunk 4 above)
Comment (Surgo) [reply packages/optimizely_x/src/index.ts]: 👍

File: packages/optimizely_x/src/plugin.ts (hunk 2 above)
Comment (Surgo) [reply packages/optimizely_x/src/plugin.ts]: 👍

**Pull request merged**